conans/client/build/cmake_flags.py

```diff
@@ -172,6 +172,61 @@ def _get_cpp_standard_vars(self):
         definitions["CONAN_STD_CXX_FLAG"] = cppstd_flag(self._conanfile.settings)
         return definitions
 
+    @staticmethod
+    def _cmake_system_name(os_, cmake_version):
+        apple_system_name = "Darwin" if cmake_version and Version(cmake_version) < Version(
+            "3.14") or not cmake_version else None
+        cmake_system_name_map = {"Macos": "Darwin",
+                                 "iOS": apple_system_name or "iOS",
+                                 "tvOS": apple_system_name or "tvOS",
+                                 "watchOS": apple_system_name or "watchOS",
+                                 "Neutrino": "QNX",
+                                 "": "Generic",
+                                 None: "Generic"}
+        return cmake_system_name_map.get(os_, os_)
+
+    @staticmethod
+    def _cmake_system_processor(os_, arch):
+        if os_ in ["Windows", "WindowsStore", "WindowsCE"]:
```

Comment:
I *think* these names are valid for Visual Studio, not for Windows systems.
mingw expects the gnu names instead of Microsoft tainted ones.

Reply:
do you have a reference? the [doc](https://cmake.org/cmake/help/v3.19/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html#variable:CMAKE_HOST_SYSTEM_PROCESSOR) says:
```
On Windows, this variable is set to the value of the environment variable PROCESSOR_ARCHITECTURE
```
it doesn't mention compilers

Reply:
I have no other reference.
But the cmake documentation mentions GNU, and mingw is a GNU compiler.

Otoh, I searched for `CMAKE_SYSTEM_PROCESSOR arm` and `CMAKE_SYSTEM_PROCESSOR aarch` on github and it looks like there is no *universal standard*.
I think we have to face the reality that there is no universally-agreed-to-standard to use as `CMAKE_SYSTEM_PROCESSOR` variable.
Like everything in the c/c++ world (except for the stl standaard), there is no standard. CMake allows everything. CMAKE_SYSTEM_PROCESSOR is just a hint.

In reality, some project will either use one kind `CMAKE_SYSTEM_PROCESSOR` mapping from processor to cmake variable and not switch between a Microsoft convention or GNU convention. Or will not switch between Windows convention or Linux convention.

It's like [the `arch` argument of `MsBuild.build`](https://docs.conan.io/en/latest/reference/build_helpers/visual_studio.html#build): there will always be a project having a different mapping (uppercase/lowercase/64 suffix).

Making this variable too hidden will make it harder for people wanting to modify it.
I'm just saying it would be nice to have an api that allows this:

```python
self._cmake.cmake_host_processor = tools.cmake_host_processor(self.settings.arch, "gnu").tolower()
```
So people can hook it.
Cci recipe writers will thank you.

And conan providing a proper default: the Microsoft one if MSVC else GNU.

Reply:
yes, I think it's correct. CMake itself doesn't care much about `CMAKE_SYSTEM_PROCESSOR` and lists it as optional.
the main source of headache is libraries checking for `CMAKE_SYSTEM_PROCESSOR`, for instance, to include proper assembler sources, such as libjpeg-turbo case.
although CMake specifies `CMAKE_SYSTEM_PROCESSOR` to have values either of `uname` or `PROCESSOR_ARCHITECTURE`, I've seen in reality libraries checking for different values, e.g. for `armv8` they may check for `armv8`, `arm64` and `aarch64`.
examples:
- [aarch64](https://github.com/libjpeg-turbo/libjpeg-turbo/blob/ccaba5d7894ecfb5a8f11e48d3f86e1f14d5a469/CMakeLists.txt#L57)
- [arm64](https://github.com/xianyi/OpenBLAS/blob/1c0b03efb42dfc829198c74391206ef4900441e6/cmake/system_check.cmake#L57)
- [armv8-a](https://github.com/hohocb/xmrig/blob/3296c0d7f3f78540cec7df98507330c8c9a0e953/cmake/cpu.cmake#L11)

Reply:
On 1.33.0 we needed this patch to make the libjpeg-turbo cross compile for "ARM64/aarch64/armv8 etc." (not saying its not libjpeg turbos fault)